blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/BlueTrendContainerImpl.java

```diff
@@ -0,0 +1,52 @@
+package io.jenkins.blueocean.service.embedded.rest;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterators;
+import io.jenkins.blueocean.commons.ServiceException;
+import io.jenkins.blueocean.rest.factory.BlueTrendFactory;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.rest.model.BluePipeline;
+import io.jenkins.blueocean.rest.model.BlueTrend;
+import io.jenkins.blueocean.rest.model.BlueTrendContainer;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Iterator;
+
+// TODO: reenable this after refactor
+// @Restricted(NoExternalUse.class)
+public class BlueTrendContainerImpl extends BlueTrendContainer {
+
+    private final BluePipeline pipeline;
+
+    public BlueTrendContainerImpl(BluePipeline pipeline) {
+        this.pipeline = pipeline;
+    }
+
+    @Override
+    public Link getLink() {
+        return pipeline.getLink().rel("trends");
+    }
+
+    @Override
+    public BlueTrend get(final String name) {
+        BlueTrend trend = Iterators.find(iterator(), new Predicate<BlueTrend>() {
+            @Override
+            public boolean apply(@Nullable BlueTrend input) {
+                return input != null && input.getId().equals(name);
```

Comment:
`BlueTrend.getId()` should give unique id, default implementation should generate a unique id so that it ensures uniqueness across implementations. We can add method to BlueTrend, `BlueTrend.getDescription()` to give human readable description?

Reply:
@vivek Based on our discussion yesterday do you think this is still required?

Reply:
@cliffmeyers Not sure, basically, if someone implements BlueTrend, what id should they be returning? Do they override default 'stageDuration' with their own implementation? Or they are allowed to provide their own trend but then they need to figure what id should they be using? Maybe I should chat you offline to get it cleared up. I will ping you my evening/your morning.